### PR TITLE
Update multiai ultrasound application for optimal performance

### PR DESCRIPTION
**Pull request by AndreasHeumann**
Use CUDA stream pool to enable operators to run in parallel on the GPU. Use BlockMemoryPool to avoid calls to cudaFree() which synchronize CPU and GPU execution.
Execute post processing on GPU (needs Holoscan SDK 2.6).

Improves performance from 12 ms to 7.4 ms per frame.

### CODE REVIEW

File: applications/multiai_ultrasound/cpp/main.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * SPDX-FileCopyrightText: Copyright (c) 2022-2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-FileCopyrightText: Copyright (c) 2022-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.nan
```
[the remark this answers is not in the document]
Comment (AndreasHeumann) [reply applications/multiai_ultrasound/cpp/main.cpp]: Thenks, yes, removed.

File: applications/multiai_ultrasound/cpp/main.cpp [hunk 2]
```diff
@@ -17,97 +17,141 @@
...
 #include <holoscan/operators/inference/inference.hpp>
 #include <holoscan/operators/inference_processor/inference_processor.hpp>
+#include <holoscan/operators/video_stream_replayer/video_stream_replayer.hpp>
+#include <holoscan/version_config.hpp>
+
 #include <visualizer_icardio.hpp>
-#include <holoscan/operators/holoviz/holoviz.hpp>
+
+#define HOLOSCAN_VERSION \
+  (HOLOSCAN_VERSION_MAJOR * 10000 + HOLOSCAN_VERSION_MINOR * 100 + HOLOSCAN_VERSION_PATCH)
```
[the remark this answers is not in the document]
Comment (AndreasHeumann) [reply applications/multiai_ultrasound/cpp/main.cpp:32]: Yes, we can file an issue on Holoscan SDK on this.

**Comment by AndreasHeumann** [comment]
@tbirdso Addressed the comments, is this now good to be merged?

**Review by tbirdso — APPROVED** [review]

**Comment by tbirdso** [comment]
Thanks @AndreasHeumann , looks good to me. Please resolve merge conflicts and ping me or @jjomier again for merge.

**Comment by AndreasHeumann** [comment]
Thanks @tbirdso , rebased the PR and resolved conflicts.

**Pull request merged**